nemo_text_processing/g2p/modules.py

```diff
@@ -496,3 +496,95 @@ def __call__(self, text):
+                It is useful to mark word as unchangeable which is already in phoneme representation.
+            apply_to_oov_word: Function that will be applied to out of phoneme_dict word.
+            ignore_ambiguous_words: Whether to not handle word via phoneme_dict with ambiguous phoneme sequences. Defaults to True.
+            heteronyms (str, Path, List): Path to file with heteronyms (every line is new word) or list of words.
+            encoding: Encoding type.
+            phoneme_probability (Optional[float]): The probability (0.<var<1.) that each word is phonemized. Defaults to None which is the same as 1.
+                Note that this code path is only run if the word can be phonemized. For example: If the word does not have a entry in the g2p dict, it will be returned
+                as characters. If the word has multiple entries and ignore_ambiguous_words is True, it will be returned as characters.
+        """
+        phoneme_dict = (
+            self._parse_as_pinyin_dict(phoneme_dict)
+            if isinstance(phoneme_dict, str) or isinstance(phoneme_dict, pathlib.Path) or phoneme_dict is None
+            else phoneme_dict
+        )
+
+        if apply_to_oov_word is None:
+            logging.warning(
+                "apply_to_oov_word=None, This means that some of words will remain unchanged "
+                "if they are not handled by any of the rules in self.parse_one_word(). "
+                "This may be intended if phonemes and chars are both valid inputs, otherwise, "
+                "you may see unexpected deletions in your input."
+            )
+
+        super().__init__(
+            phoneme_dict=phoneme_dict,
+            word_tokenize_func=word_tokenize_func,
+            apply_to_oov_word=apply_to_oov_word,
+            mapping_file=mapping_file,
+        )
+        self.tones = {'1': '#1', '2': '#2', '3': '#3', '4': '#4', '5': '#5'}
+        from pypinyin import lazy_pinyin, Style
+
+        self._lazy_pinyin = lazy_pinyin
+        self._Style = Style
+
+    def _parse_as_pinyin_dict(self, phoneme_dict_path):
+        """Loads pinyin dict file, and generates a set of all valid symbols."""
+        g2p_dict = defaultdict(list)
+        with open(phoneme_dict_path, 'r') as file:
+            for line in file:
+                parts = line.split('\t')
+                # let the key be lower, since pypinyin would give lower representation
+                pinyin = parts[0].lower()
+                pronunciation = parts[1].split()
+                pronunciation_with_sharp = ['#' + pron for pron in pronunciation]
+                g2p_dict[pinyin] = pronunciation_with_sharp
+        return g2p_dict
+
+    def __call__(self, text):
+        """
+        errors func handle below is to process the bilingual situation,
+        specifically, it would return words into letters.
+        e.g. 我今天去了Apple Store, 买了一个ihone。
```

Comment:
it seems that individual English letters are used as "phonemes", while g2p did not look English words up in the English dict to get English phonemes. So the synthesized audio of English words is just like letter sound, right? 

If so, is there any way to improve it by looking up our English dict?

Reply:
Not exactly. For Chinese, the model uses phone as modeling unit. While for English, the model will receive grapheme representation of text (letters) as input. What's the synthesized audio sounds like would depend on the ground truth audio. E.g. for "cat", it could be "c ae t" or "c a t".

It is possible to use phone for English modeling unit. However, I chose to use letters for code simplification, since we need to combine pureChinesePhoneTokenier with pureEnglishPhoneTokenier into a bilingual Tokenier. (EnglishPhoneTokenier is much more complicated than EnglishCharTokenier)

Reply:
> What's the synthesized audio sounds like would depend on the ground truth audio. E.g. for "cat", it could be "c ae t" or "c a t".

understood. We are good for now. The only concern is that parallel English letters and audio clips are rather scarce in the bilingual setup compared to the rich Chinese data resources. Look forward to the listening test and verifying it further.

Reply:
Yes. I think for this bilingual dataset, it would be better to start from a LJspeech pretrained model which covers more English. Maybe consider to add a notebook tutorial for this two stage training (LJspeech -> SF_bilingual) in next PR.

I would prepare some audios and ground truth audios for check.